OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs

```diff
@@ -120,8 +120,7 @@ public GameInfoStatsLogic(Widget widget, World world, OrderManager orderManager,
 						item.Get<LabelWidget>("FACTION").GetText = () => pp.DisplayFaction.Name;
 					}
 
-					var experience = p.Second != null ? p.Second.Experience : 0;
-					item.Get<LabelWidget>("SCORE").GetText = () => experience.ToString();
+					item.Get<LabelWidget>("SCORE").GetText = () => (p.Second != null ? p.Second.Experience : 0).ToString();
```

Comment:
If you don't mind the scope creep we could save some memory churn here by using `CachedTransform` to only allocate new strings when the experience actually changes.  It would be good to expand this to the other cells too.

Reply:
erm... I'm probably doing it wrong, but it's not working:

```diff
--- a/OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs
+++ b/OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs
@@ -9,6 +9,7 @@
  */
 #endregion
 
+using System.Collections.Generic;
 using System.Drawing;
 using System.Linq;
 using OpenRA.Graphics;
@@ -77,8 +78,9 @@ public GameInfoStatsLogic(Widget widget, World world, OrderManager orderManager,
                                        var teamHeader = ScrollItemWidget.Setup(teamTemplate, () => true, () => { });
                                        teamHeader.Get<LabelWidget>("TEAM").GetText = () => t.Key == 0 ? "No Team" : "Team {0}".F(t.Key);
                                        var teamRating = teamHeader.Get<LabelWidget>("TEAM_SCORE");
-                                       teamRating.GetText = () => t.Sum(gg => gg.Second != null ? gg.Second.Experience : 0).ToString();
-
+                                       var teamScore = new CachedTransform<IGrouping<int, Pair<Player, PlayerStatistics>>, string>(p =>
+                                               p.Sum(gg => gg.Second != null ? gg.Second.Experience : 0).ToString());
+                                       teamRating.GetText = () => teamScore.Update(t);
                                        playerPanel.AddChild(teamHeader);
                                }
```

Reply:
Updated